frontend/components/PasteHandler.js

```diff
@@ -0,0 +1,101 @@
+import { html, useEffect } from "../imports/Preact.js"
+import { link_open_path } from "./Welcome.js"
+
+const detectNotebook = (inputtext) => {
+    // Add a newline in the end for the case user didn't copy it
+    // That helps if the user copied up to the last line of the cell order
+    const text = `${inputtext}\n`
+    const from = text.indexOf("### A Pluto.jl notebook ###")
+    const cellsfound = text.match(/# ... ........-....-....-....-............/g)
+    const cellscount = cellsfound?.length
+    const cellsorder = text.indexOf("# ╔═╡ Cell order:") + "# ╔═╡ Cell order:".length + 1
+    let i = 0
+    let to = cellsorder
+
+    while (++i <= cellscount) {
+        to = text.indexOf("\n", to + 1) + 1
```

Comment:
Can this be a `for` loop? I spent some time figuring out whether this could get stuck in an infinite loop until I realised that it's just a for loop